### DISCUSSION TITLE
Is this repository faster than sklearn.metrics?

### DISCUSSION THREAD

**Discussion by mjack3** [Lightning-AI/torchmetrics#1066] [General]
Hello. I have just discovered this repository right now and it seems very interesting. I wonder what's the difference respect using. As fat as I Know this work have a pretty long metrics available but in terms of speed, did you tested it?

I couldn't find nothing in your documentation :). It would be great to describe the advantages with respect using classic tools.

**Comment by aiqc** [comment]
not so much a question of speed. it's about data format (torch vs numpy), dimensionality (2D+?, how binary/multi classify are handled), and compatibility (trainer history metrics). 

i will say that torchmetrics is more comprehensive than keras metrics (e.g. F1 score), but you don't have to do any data wrangling w keras metrics

**Comment by SkafteNicki** [comment]
We have not tested if we are actually faster than `sklearn`. However, I can confidently say that it dependents on the situation:
* If you are doing calculations on cpu with reasonable size input, I highly suspect that `sklearn` is faster
* If you are doing calculations on gpu, we should be faster just based on the fact that if you have do move all data to cpu constantly to use `sklearn` 

As @aiqc also states we focus more on fitting our metrics into a deep learning perspective where I would say `sklearn` fits a more general machine learning perspective. One thing that `sklearn` cannot handle is accumulation over multiple batches, which `torchmetrics` can deal with because that is how data is processed in a deep learning pipeline.

**Comment by mjack3** [comment]
Hello. Just wondering if this repository supports GPU metric computation :).
